test/python/circuit/test_uc.py

```diff
@@ -100,6 +107,21 @@ def test_inverse_ucg(self):
 
         self.assertTrue(np.allclose(unitary_desired, unitary))
 
+    def test_ucge(self):
```

Comment:
why do you need a specific test? Isn't it enough to add this specific `gate_list` of hadamards to `squs` above?